### PR TITLE
 Simple ATM Bank in C

### LINKED ISSUE #3159: Simple ATM Bank in C
### Description

A simple program for beginners to choose at the initial stage to make a project in C! 👍
The program will be a menu driven program which will allow the user to-
- [x] Check the balance 
- [x] Money deposited
- [x] Money withdraw

Hope you understand the simple code and code it in you system! 💯 
Enjoyyy coding! 👍 

### Domain

C/CPP

### Type of Contribution

Documentation

### Code of Conduct

- [X] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.

### PR DESCRIPTION
**Pull request by ikavyajain**
<hr>

## Description 📜

Simple ATM Bank in C
A simple program for beginners to choose at the initial stage to make a project in C! 👍
The program will be a menu driven program which will allow the user to-

- [x] Check the balance
- [x] Money deposited
- [x] Money withdraw

Hope you understand the simple code and code it in you system! 💯
Enjoyyy coding! 👍

Fixes #3159

<hr>

## Type of change 📝

- [x] Documentation (Content Creation in the form of codes or tutorials)

<hr>

- [x] C/CPP

<hr>

## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

<hr>

## Screenshots / Gif (Optional) 📸

GAME1
GAME2
GAME3
GAME4

<hr>

### CODE REVIEW

**Review by vasu-1 — CHANGES_REQUESTED** [review]
Good Effort @ikavyajain 

But your code is having some bug !
**when i try to give deposite two time than it is giving me an error**

try to figure it out  !!!

**Comment by vasu-1** [comment]
Also Remove all this line ` system("CLS"); ` from your code !!!

**Comment by ikavyajain** [comment]
Just try to run on turbo!
Because i have coded on it and it was kinda successful!

**Comment by vasu-1** [comment]
> Just try to run on turbo! Because i have coded on it and it was kinda successful!

Suggestion : Use Modern Compilers like **ming-gw**

**Comment by vasu-1** [comment]
> Just try to run on turbo! Because i have coded on it and it was kinda successful!

I think yes!!
It is working fine

File: C_CPP/Competitive Programming/Simple_ATM_Bank_in_C.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+                                                      float moneyDeposit(float balance);
+                                                      float moneyWithdraw(float balance);
+                                                      void menuExit();
+                                                      void errorMessage();
+                                                      int main()
+                                                      {
+                                                      int option;
+                                                      float balance = 0.00;
+                                                      int choose;
+                                                      bool again = true;
```
Comment (vasu-1) [inline C_CPP/Competitive Programming/Simple_ATM_Bank_in_C.md:21]:
```suggestion
```c++
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>
#include <math.h>

void login();
void mainMenu();
void checkBalance(float balance);
float moneyDeposit(float balance);
float moneyWithdraw(float balance);
void menuExit();
void errorMessage();

int main()
{
   int option;
   float balance = 0.00;
   int choose;
   bool again = true;
```

do this to your all code !

File: C_CPP/Competitive Programming/Simple_ATM_Bank_in_C.md [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+                                                      void menuExit() {
+                                                      printf("--------------Take your receipt!!!------------------\n");
+                                                      printf("-----Thank you for using ATM Banking Machine!!!-----\n");
+                                                      }
+                                                      //exit menu
+                                                      void errorMessage() 
+                                                      {;
+                                                      printf("+++!!!You selected invalid number!!!+++\n");
+                                                      }
+
```
Comment (vasu-1) [inline C_CPP/Competitive Programming/Simple_ATM_Bank_in_C.md:122]:
```suggestion
{;
   printf("+++!!!You selected invalid number!!!+++\n");
}
```
                                                      
```

**Review by vasu-1 — CHANGES_REQUESTED** [review]
Okk please do these changes
- use some markdown
- explain code in the comments
- remove those all spaces from the code and put apropriate spaces.

**Comment by ikavyajain** [comment]
Made the required changes! Do check

**Comment by Suvraneel** [comment]
@vasu-1 Review

@ikavyajain  Btw turbo-C & other DOS based compilers are now obsolete
So, graphics.h & similar headers are deprecated...
Try to use ming-gw64... gcc etc...

**Comment by ikavyajain** [comment]
Okay i’ll take care of that from now on!

File: C_CPP/Competitive Programming/Simple_ATM_Bank_in_C.md [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+                                                            printf("-----Thank you for using ATM Banking Machine!!!-----\n");
+                                                      }
+                                                      
+                                                      
+                                                      //exit menu
+                                                      void errorMessage() 
+                                                      {;
+                                                      printf("+++!!!You selected invalid number!!!+++\n");
+                                                      }
+
```
Comment (vasu-1) [inline C_CPP/Competitive Programming/Simple_ATM_Bank_in_C.md:135]:
```suggestion
```cpp
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>
#include <math.h>
void login();
void mainMenu();
void checkBalance(float balance);    // To check balance in the account
float moneyDeposit(float balance);   // To check the amount deposited in the account
float moneyWithdraw(float balance);  // To check the amount withdrawed from the account
void menuExit();                     // To quit the menu
void errorMessage();
int main()
{
  int option;
  float balance = 0.00;
  int choose;
  bool again = true;
  while (again)
  {
    mainMenu();
    printf("=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-\n");
    printf("Your Selection:\t");
    scanf("%d", &option);
    switch (option)
    {
      case 1:
      checkBalance(balance);           	                    
      break;
      case 2:
      balance = moneyDeposit(balance);            	                                                                                                                                   break;
      case 3:           	                                       
      balance = moneyWithdraw(balance);
      break;
      case 4:            	                                        
      menuExit();
      return 0;
      default:
      errorMessage();
      break;
    }

    printf("=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=\n");
    printf("Would you like to do another transaction:\n");
    printf("< 1 > Yes\n");
    printf("< 2 > No\n");
    scanf("%d", &choose);

    if (choose == 2)
    {
      again = false;
      menuExit();
    }
  }
  return 0;
}
                                                  
                                                  
void mainMenu()
{
  printf("**********Welcome to ATM Bank***********\n\n");
  printf("****Please choose one of the options below****\n\n");
  printf("< 1 >  Check Balance\n");
  printf("< 2 >  Deposit\n");
  printf("< 3 >  Withdraw\n");
  printf("< 4 >  Exit\n\n");
}
                                                  
                                                  
//Main Menu
void checkBalance(float balance)
{
  printf("You Choose to See your Balance\n");
  printf("\n\n****Your Available Balance is:   $%.2f\n\n", balance);
}
                                                  
                                                  
//Check Balance
float moneyDeposit(float balance) 
{
  float deposit;
  printf("You choose to Deposit a money\n");
  printf("Your Balance is: $%.2f\n\n", balance);
  printf("****Enter your amount to Deposit\n");
  scanf("%f", &deposit);
  balance += deposit;
  printf("\n****Your New Balance is:   $%.2f\n\n", balance);

  return balance;
}
                                                  
                                                  
//money deposit
float moneyWithdraw(float balance) 
{
  float withdraw;
  bool back = true;
  printf("You choose to Withdraw a money\n");
  printf("Your Balance is: $%.2f\n\n", balance);
  while (back) 
  {
    printf("Enter your amount to withdraw:\n");
    scanf("%f", &withdraw);
    if(withdraw < balance)
    {
      back = false;
      balance -= withdraw;
      printf("\nYour withdrawing money is:  $%.2f\n", withdraw);
      printf("****Your New Balance is:   $%.2f\n\n", balance);
    }
    else  
    {
      printf("+++You don't have enough money+++\n");
      printf("Please contact to your Bank Customer Services\n");
      printf("****Your Balance is:   $%.2f\n\n", balance);
    }
  }
  return balance;
}

//money withdraw
void menuExit() 
{
  printf("--------------Take your receipt!!!------------------\n");
  printf("-----Thank you for using ATM Banking Machine!!!-----\n");
}

//exit menu
void errorMessage() 
{
  printf("+++!!!You selected invalid number!!!+++\n");
}

```
```

Comment (vasu-1) [inline C_CPP/Competitive Programming/Simple_ATM_Bank_in_C.md:135]:
```suggestion
```cpp
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>
#include <math.h>
void login();
void mainMenu();
void checkBalance(float balance);    // To check balance in the account
float moneyDeposit(float balance);   // To check the amount deposited in the account
float moneyWithdraw(float balance);  // To check the amount withdrawed from the account
void menuExit();                     // To quit the menu
void errorMessage();
int main()
{
  int option;
  float balance = 0.00;
  int choose;
  bool again = true;
  while (again)
  {
    mainMenu();
    printf("=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-\n");
    printf("Your Selection:\t");
    scanf("%d", &option);
    switch (option)
    {
      case 1:
      checkBalance(balance);           	                    
      break;
      case 2:
      balance = moneyDeposit(balance);
      break;
      case 3:           	                                       
      balance = moneyWithdraw(balance);
      break;
      case 4:            	                                        
      menuExit();
      return 0;
      default:
      errorMessage();
      break;
    }

    printf("=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=\n");
    printf("Would you like to do another transaction:\n");
    printf("< 1 > Yes\n");
    printf("< 2 > No\n");
    scanf("%d", &choose);

    if (choose == 2)
    {
      again = false;
      menuExit();
    }
  }
  return 0;
}
                                                  
                                                  
void mainMenu()
{
  printf("**********Welcome to ATM Bank***********\n\n");
  printf("****Please choose one of the options below****\n\n");
  printf("< 1 >  Check Balance\n");
  printf("< 2 >  Deposit\n");
  printf("< 3 >  Withdraw\n");
  printf("< 4 >  Exit\n\n");
}
                                                  
                                                  
//Main Menu
void checkBalance(float balance)
{
  printf("You Choose to See your Balance\n");
  printf("\n\n****Your Available Balance is:   $%.2f\n\n", balance);
}
                                                  
                                                  
//Check Balance
float moneyDeposit(float balance) 
{
  float deposit;
  printf("You choose to Deposit a money\n");
  printf("Your Balance is: $%.2f\n\n", balance);
  printf("****Enter your amount to Deposit\n");
  scanf("%f", &deposit);
  balance += deposit;
  printf("\n****Your New Balance is:   $%.2f\n\n", balance);

  return balance;
}
                                                  
                                                  
//money deposit
float moneyWithdraw(float balance) 
{
  float withdraw;
  bool back = true;
  printf("You choose to Withdraw a money\n");
  printf("Your Balance is: $%.2f\n\n", balance);
  while (back) 
  {
    printf("Enter your amount to withdraw:\n");
    scanf("%f", &withdraw);
    if(withdraw < balance)
    {
      back = false;
      balance -= withdraw;
      printf("\nYour withdrawing money is:  $%.2f\n", withdraw);
      printf("****Your New Balance is:   $%.2f\n\n", balance);
    }
    else  
    {
      printf("+++You don't have enough money+++\n");
      printf("Please contact to your Bank Customer Services\n");
      printf("****Your Balance is:   $%.2f\n\n", balance);
    }
  }
  return balance;
}

//money withdraw
void menuExit() 
{
  printf("--------------Take your receipt!!!------------------\n");
  printf("-----Thank you for using ATM Banking Machine!!!-----\n");
}

//exit menu
void errorMessage() 
{
  printf("+++!!!You selected invalid number!!!+++\n");
}

```
```

**Review by vasu-1 — CHANGES_REQUESTED** [review]
@ikavyajain 
do like this !!!

**Comment by ikavyajain** [comment]
Changes are made properly, this time! 
Hope this looks much better now! :)

**Review by vasu-1 — APPROVED** [review]
Approved ⭐✨

Good to go for merge @Suvraneel

**Comment by Suvraneel** [comment]
@ikavyajain  LGTM

**Pull request merged**